### ISSUE TITLE
storing the best model

### ISSUE THREAD

**Issue by saikrishna-1996** [pytorch/examples#232]
https://github.com/pytorch/examples/blob/7d0d413425e2ee64fcd0e0de1b11c5cca1f79f4d/word_language_model/main.py#L171
doesn't look like it is actually saving the best model? It saves only the current model

**Comment by DNXie** [comment]
Same problem.  
Is there anything like a `keras.callbacks.ModelCheckpoint` that saves only the best model after each epoch? Or we just have to write a `for` loop and compare it ourselves.

**Comment by nzmora** [comment]
The code saves the current model if the current epoch produced a model that has a lower validation loss than the best model seen so far.  So in the way this particular example is coded (and you can surely change it), you end up with only one saved model, which captures the state of the best model seen during the entire training session ("best" is as measured by comparing validation loss; at the end of each training epoch).

Cheers,
Neta

**Comment by liu09114** [comment]
Had you solve this problom?
@saikrishna-1996

**Comment by nzmora** [comment]
@saikrishna-1996 ,

The code is not broken, it operates as explained above.  I see that you gave [this explanation](https://github.com/pytorch/examples/issues/232#issuecomment-395198394) a -1, which doesn't help in this case because it is not clear what you don't like about the implemented behavior and it would help if you gave some detail.

Thanks
Neta

**Comment by BIRAN** [comment]
Had you solved this problem?
@saikrishna-1996
